src/utils/AppRouteProvider.jsx

```diff
@@ -43,7 +45,19 @@ const AppRouteProvider = (props) => {
   const router = useRouter();
   const dispatch = useDispatch();
 
+  const pathLookup = useRef(Object.entries(PATH_STUBS));
+
   const [renderIntercept, setRenderIntercept] = useState(null);
+  const [currentModule, setCurrentModule] = useState(module.DATA_MANAGEMENT);
+
+  useEffect(() => {
+    pathLookup.current.find(([moduleName, path]) => {
```

Comment:
the `find` function returns the first element that matches the condition but you are ignoring the return value and instead doing the logic inside the condition matching. You should do instead:

```
const moduleName = pathLookup.find(([mn, p]) => p === '/data-processing')[0]
setCurrentModule(moduleName);
```

and I don't think you need to use `.match` which is designed to regex when you are just comparing normal strings.

Reply:
the value of `router.pathname` is the real pathname (e.g. `http://localhost:5000/experiments/8be39ad79c4f16e611650f8c90a4d3f6/data-processing`). That's why we need to match the path.

I purposefully use `Array.find()` so that it returns on the first instance of `true`, which is when the pathname matches the path and the lookup doesn't have to continue.

Reply:
changed